src/machines/builder_route/axis.config.js

```diff
@@ -0,0 +1,140 @@
+export const axis_config = {
+	id: "Axis",
+	initial: "CheckingAxis",
+	on: {
+		'SELECTION_CHANGED': {
+			target: "Axis.CheckingAxis",
+			actions: ['setURL']
+		},
+		"AXIS_RESET": {
+			target: "Axis.AxisIncomplete",
+			actions: [
+				'deleteChildAxes',
+				'setURL'
+			]
+		}
+	},
+	states: {
+		"CheckingAxis": {
+			id: "CheckingAxis",
+			on: {
+				'': [
+					{
+						target: 'AxisComplete',
+						cond: 'isAxisComplete',
+						actions: ['spawnChildAxis']
+					},
+					{ target: 'AxisIncomplete' }
+				]
+			}
+		},
+		"AxisIncomplete": {
+			id: "AxisIncomplete",
+		},
+		"AxisComplete": {
+			id: "AxisComplete",
+			initial: "CheckingQuery",
+			on: {
+				"QUERY_CHANGED": {
+					target: "AxisComplete.CheckingQuery",
+					actions: ['setURL']
+				}
+			},
+			states: {
+				"CheckingQuery": {
+					id: "CheckingQuery",
+					on: {
+						'': [
+							{
+								target: 'QueryReady',
+								cond: 'isQueryReady'
+							},
+							{ target: 'QueryNotReady' }
+						]
+					}
+				},
+				"QueryNotReady": {
+					id: "QueryNotReady",
+				},
+				"QueryReady": {
+					id: "QueryReady",
+					initial: "CheckMatching",
+					states: {
+						"CheckMatching": {
+							onEntry: ["verifyMatching"],
+							on: {
+								'': [
+									{
+										target: 'Matching',
+										cond: 'isMatching'
+									},
+									{ target: 'Dirty' }
+								]
+							}
+						},
+						"Matching": {
+							id: "Matching",
+							onEntry: [
+								"notifyMatching",
+								"pushHistory"
+							]
+						},
+						"Dirty": {
+							id: "Dirty",
+							initial: "Idle",
+							onEntry: "notifyDirty",
+							on: {
+								"QUERY_EXECUTED": {
+									target: "Dirty.CheckingCache"
+								}
+							},
+							states: {
+								"Idle": {
+									on: {
+										'': [
+											{
+												target: 'CheckingCache',
+												cond: 'isAutoExecute'
+											}
+										]
+									}
+
+								},
+								"CheckingCache": {
+									onEntry: ["searchInCache"],
+									on: {
+										'': [
+											{
+												target: '#Matching',
+												cond: 'isInCache'
+											},
+											{ target: 'Pending' }
+										]
+									}
+								},
+								"Pending": {
+									onEntry: "notifyPending",
```

Comment:
If you're refering to the `notifyPending` action, I was going to ask about that today.

Though it was originally called `sharePending` I figured the actions were originally used to set flags to guide the UI so I left them there tentatively at key moments in the state flow. 

Perhaps their intended purpose was another so if they turn out to be useless I'll remove them.

Reply:
Those actions are used in HelathMosaic to communicate with the parent (`sharePending: sendParent('PENDING'),`) because search machines are spawned in that case, but that doesn't apply here.

Reply:
In this case however, axis machines will be spawned. Will they need to communicate with the parent?

Reply:
Not that I know of with the current features set? We can add new actions hen the need will arise I think.